### ISSUE TITLE
govc: Moving a new host portgroup to folder

### ISSUE THREAD

**Issue by lukasluecke** [vmware/govmomi#2108]
I am calling `host.portgroup.add` first and then `object.mv` afterwards. This works most of the time, but sometimes (can't reliably reproduce it, but it happened often enough to be a concern) it fails because it can't find the newly created portgroup. 

```
The command "'govc' 'object.mv' './network/pg-d-dev_l-A1-50j3v-p-lkrlk-lan2' './network/vs-d-dev_l/vs-d-dev_l-A1-50j3v-p-lkrlk'" failed.

Exit Code: 1(General error)

Error Output:
================
govc: object './network/pg-d-dev_l-A1-50j3v-p-lkrlk-lan2' not found"
```

Trying to run the exact same command manually afterwards works, so it looks to be a timing issue. Any ideas or suggestions on how to solve or work around this? Would it be possible to add `FolderFlags` to this command (like `vm.clone`) maybe, so it can be handled in one go which might be less prone to this issue?

**Comment by dougm** [comment]
Not sure, but it could be the API to create the host portgroup returns before vCenter adds it to the inventory folder.
You could do the following before calling mv, which will wait (`5m` max) until a network with the given name exists in the network inventory folder: 
```bash
govc object.collect -s -wait 5m -type n ./network -name pg-d-dev_l-A1-50j3v-p-lkrlk-lan2
```

**Comment by lukasluecke** [comment]
Seems to work, thank you! 🙂

However sometime the wait time seems to be too long - I specified `10s` for example, but it waits for 20s 🤔 I will play with some different settings, and see if I can figure out what's happening.

**Comment by dougm** [comment]
Not quite seeing that here, but there are API calls, etc., that happen before the `-wait` duration takes effect. How does it look for you with a longer `-wait` time?

**Comment by lukasluecke** [comment]
Okay, wait time seems to be much closer to the expected value at `30s`.

The problem is that it always waits the full amount instead of returning once the element exists.
Starting another `object.collect` call a few seconds later (after the object was created) instantly returns, while the first one is still waiting.

**Comment by dougm** [comment]
You're right. The `-prop val` filter works well to wait on an existing object field to change, but seems not to wait for a new object. Can look into making that work, as it'd be useful. But in the meantime, another way that should work for you:

```bash
govc object.collect -n 1 ./network childEntity >/dev/null & # this command will exit when a new network is added
govc host.portgroup.add ...
wait # wait for object.collect to exit
govc object.mv ...
```

**Comment by dougm** [comment]
@lukasluecke see the govc change in https://github.com/vmware/govmomi/pull/2141 , made it part of that as the vcsim fix was required to test the govc fix... let me know if you can give it a try.

**Comment by dougm** [comment]
Merged #2141 , so the original suggestion (below) should work now, feel free to re-open if that isn't the case.
```
govc object.collect -s -wait 5m -type n ./network -name pg-d-dev_l-A1-50j3v-p-lkrlk-lan2
```
